src/Pagination/components/PageNumberButton.ts

```diff
@@ -1,22 +1,52 @@
-import { SFC, createElement } from "react";
+import { ReactElement, createElement } from "react";
 import * as classNames from "classnames";
 
 interface PageNumberButtonProps {
     pageNumber: number;
     selectedPageNumber: number;
-    onClickAction: (pageNumber: number) => void;
+    onClickAction: () => void;
     key?: string | number;
 }
 
-export const PageNumberButton: SFC<PageNumberButtonProps> = (props: PageNumberButtonProps) => createElement("li", {
-        className: classNames(
-            props.selectedPageNumber === props.pageNumber ? "active" : "",
-            props.pageNumber < 10 ? "single-digit" : ""
-        ),
-        onClick: () => props.onClickAction(props.pageNumber),
-        key: props.key
-    },
-    props.pageNumber
-);
+export const PageNumberButton = (
+    props: PageNumberButtonProps
+): ReactElement => {
+    const { onClickAction, pageNumber, selectedPageNumber } = props;
+
+    return createElement(
+        "li",
+        {
+            className: classNames(
+                selectedPageNumber === pageNumber ? "active" : ""
+            ),
+            role: "button",
+            onClick: onClickAction,
+            onKeyDown: onKeyDown.bind(null, onClickAction),
+            onKeyUp: onKeyUp.bind(null, onClickAction),
+            key: props.key,
+            tabindex: 0,
+            "aria-label":
+                selectedPageNumber === pageNumber
+                    ? `Current page, page ${pageNumber}`
+                    : `Go to page ${pageNumber}`
+        },
+        pageNumber
+    );
+};
+
+const onKeyDown = (onClickAction: () => void, e: KeyboardEvent) => {
+    // if enter key
+    if (e.keyCode === 13) {
+        onClickAction();
+    }
+};
+
+const onKeyUp = (onClickAction: () => void, e: KeyboardEvent) => {
```

Comment:
Lets use only `onKeyDown`

Reply:
Initially I only used keydown according to the note in the following spec from mdn: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role. However, when I checked browser behaviour Chrome, Firefox and IE11 all wait for space keyup to execute the onclick. That's why I separated them.

Reply:
Only using `onKeyDown` now